app/src/test/java/com/codesoom/assignment/auth/application/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,148 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(MockitoExtension.class)
+class AuthenticationServiceTest {
+    private static final String SECRET = "12345678901234567890123456789010";
+    private static final String GIVEN_USER_EMAIL = "test@test.com";
+    private static final String GIVEN_USER_PASSWORD = "password";
+    private static final String NOT_EXIST_EMAIL = "not_exist@test.com";
+    private static final String WRONG_PASSWORD = "wrong_password";
+    private static final Long GIVEN_ID = 1L;
+    private static final long EXPIRED_TIME = 300000;
+    private static final String VALID_TOKEN = "eyJhbGciOiJIUzI1NiJ9.eyJ1c2VySWQiOjEsImlhdCI6MTY0MDk2MjgwMCwiZXh" +
+            "wIjoxNjQwOTYzMTAwfQ.2siRnBJmRU2JXjZY0CkQMgnCHRJN4Dld4_wG6R7T-HQ";
+
+    private AuthenticationService authenticationService;
+
+    @Mock
+    private UserRepository userRepository;
+
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        JwtTokenProvider jwtTokenProvider = new JwtTokenProvider(SECRET, EXPIRED_TIME);
+        ReflectionTestUtils.setField(jwtTokenProvider,
+                "time", LocalDateTime.of(2022, Month.JANUARY, 1, 0, 0));
+        authenticationService = new AuthenticationService(jwtTokenProvider, userRepository);
+
+        user = User.builder()
+                .id(GIVEN_ID)
+                .password(GIVEN_USER_PASSWORD)
+                .email(GIVEN_USER_EMAIL)
+                .build();
+    }
+
+    @Nested
+    @DisplayName("authenticate 메서드는")
+    class Describe_authenticate {
+
+        @Nested
+        @DisplayName("등록된 이메일과 비밀번호가 주어지면")
+        class Context_with_exist_email_password {
+            String email;
+            String password;
+
+            @BeforeEach
+            void setUp() {
+                email = GIVEN_USER_EMAIL;
+                password = GIVEN_USER_PASSWORD;
+
+                given(userRepository.findByEmail(GIVEN_USER_EMAIL))
+                        .willReturn(Optional.of(user));
+            }
+
+            @DisplayName("인증 토큰을 리턴한다.")
+            @Test
+            void it_returns_token() {
+                String token = authenticationService.authenticate(email, password);
+                assertThat(token).isEqualTo(VALID_TOKEN);
+            }
+        }
+
+        @Nested
+        @DisplayName("등록되지 않은 이메일이 주어지면")
+        class Context_with_not_exist_email {
+            String email;
+            String password;
+
+            @BeforeEach
+            void setUp() {
+                email = NOT_EXIST_EMAIL;
+                password = GIVEN_USER_PASSWORD;
+            }
+
+            @DisplayName("예외를 발생시킨다.")
```

Comment:
이건 미묘한 차이이긴 한데요, 예외는 throw. 우리가 작성한 코드가 예외를 던지죠.

```suggestion
            @DisplayName("예외를 던진다.")
```

Reply:
발생시킨다를 능동으로 생각하고 작성했는데, 지금보니 수정하는게 좋겠습니다.  
코드 작성할때는 눈치채지 못했네요 👍